### PR TITLE
Add background task to collect Cockroach zone node IDs

### PR DESCRIPTION
**Pull request by jgallagher**
As discussed in yesterday's update huddle - this adds an endpoint to the `cockroach-admin` server which internally uses the undocumented `crdb_internal.node_id()` function to get the node ID of its local running `cockroach`, and adds a background task to Nexus to collect zone ID <-> node ID mappings for every cockroach zone in the current target blueprint.

This is a prerequisite for decommissioning, but currently has no consumers.

### CODE REVIEW

File: nexus/src/app/background/crdb_node_id_collector.rs [hunk 1]
```diff
@@ -0,0 +1,573 @@
...
+                        dataset: nexus_types::inventory::OmicronZoneDataset {
+                            pool_name: format!("oxp_{}", Uuid::new_v4())
+                                .parse()
+                                .unwrap(),
+                        },
+                    },
+                ),
+            };
+
+        // Add a three CRDB zones with known addresses; the first and third are
```
Comment (andrewjstone) [inline nexus/src/app/background/crdb_node_id_collector.rs:285]:
```suggestion
        // Add three CRDB zones with known addresses; the first and third are
```

File: nexus/src/app/background/crdb_node_id_collector.rs [hunk 2]
```diff
@@ -0,0 +1,573 @@
...
+        let (_tx_blueprint, rx_blueprint) =
+            watch::channel(Some(Arc::new((blueprint_target, blueprint))));
+        let mut collector =
+            CockroachNodeIdCollector::new(datastore.clone(), rx_blueprint);
+
+        // We'll send in three Cockroach nodes for the collector to gather:
+        //
+        // 1. Node 1 will succeed
+        // 2. Node 2 will fail
+        // 3. Node 3 will succeed, but will report an unexpeted zone ID
```
Comment (andrewjstone) [inline nexus/src/app/background/crdb_node_id_collector.rs:462]:
```suggestion
        // 3. Node 3 will succeed, but will report an unexpected zone ID
```
```suggestion
        // 3. Node 3 will succeed, but will report an unexpeted zone ID
```

**Review by andrewjstone — APPROVED** [review]
Looks great.

File: cockroach-admin/src/context.rs [hunk 3]
```diff
@@ -3,7 +3,143 @@
...
+                    "err" => InlineErrorChain::new(&e),
+                );
+            }
+        });
+
+        // This uses an undocumented internal function - not awesome, but we're
+        // told this is "unlikely to change for some time".
+        // https://github.com/cockroachdb/cockroach/issues/124988 requests that
+        // this be documented (or an alternative be documented / provided).
+        let row = client
```
[the remark this answers is not in the document]
Comment (jgallagher) [reply cockroach-admin/src/context.rs:94]: Not sure if you're being too paranoid, but I _think_ this is easy enough to check for. I took a stab at it in 70c75586e

[the remark this answers is not in the document]
Comment (andrewjstone) [reply cockroach-admin/src/context.rs:94]: Thank you! This looks excellent.

**Pull request merged**